internal/broker/broker.go

```diff
@@ -636,6 +631,40 @@ func (b *Broker) handleIsAuthenticated(ctx context.Context, session *session, au
 	return AuthGranted, userInfoMessage{UserInfo: authInfo.UserInfo}
 }
 
+// userNameIsAllowed checks whether the user's username is allowed to access the machine.
+func (b *Broker) userNameIsAllowed(userName string) bool {
+	// The user is allowed to login if:
+	// - ALL users are allowed
+	// - the user's name is in the list of allowed_users
+	// - the user is the owner of the machine
```

Comment:
I disagree, explaining the intend of a function in a comment doesn't only make it easier for reviewers but also for whoever else reads the code in the future.

Reply:
I personally don't care much of comments when commit messages explain the rationale well enough (really, I comment-less code is more than fine when `git blame` can be your friend), but given this is _not_ the case (that implies: please be more elaborated in them), I feel it's better to keep it too.

Reply:
note: meta discussion which is not relevant to this PR

> I personally don't care much of comments when commit messages explain the rationale well enough (really, I comment-less code is more than fine when git blame can be your friend)

I think commit messages should explain the rationale in more detail than code comments, but IMO they don't replace code comments, because it requires significantly more work to view the commit message which originally introduced a change when the code has been modified many times since.